src/test/java/subway/line/LineAcceptanceTest.java

```diff
@@ -2,137 +2,252 @@
+
         //then
-        assertThat(lines).hasSize(2);
-        assertThat(lines).containsExactly(신분당선.get("name"), 분당선.get("name"));
+        LineStep.지하철노선을_확인(Arrays.asList(신분당선, 분당선));
     }
 
     /**
      * Given 지하철 노선을 생성하고
      * When 생성한 지하철 노선을 조회하면
      * Then 생성한 지하철 노선의 정보를 응답받을 수 있다.
-     * */
+     */
     @DisplayName("지하철노선을 조회한다.")
     @Test
     void searchLine() {
 
         //given
-        long id = saveLine(신분당선).jsonPath().getLong("id");
+        Long id = getId(LineStep.지하철노선을_추가(Arrays.asList(신분당선)));
+
         //when
-        String name = Extractor.get(LINE_URL + "/" + id).jsonPath().get("name");
+
         //then
-        assertThat(name).isEqualTo(신분당선.get("name"));
+        LineStep.지하철노선을_확인(id, 신분당선);
     }
 
     /**
      * Given 지하철 노선을 생성하고
      * When 생성한 지하철 노선을 수정하면
      * Then 해당 지하철 노선 정보는 수정된다
-     * */
+     */
     @DisplayName("지하철노선을 수정한다.")
     @Test
     void modifyLine() {
 
         //given
-        long id = saveLine(신분당선).jsonPath().getLong("id");
+        Long id = getId(LineStep.지하철노선을_추가(Arrays.asList(신분당선)));
+
         //when
-        LineResponse lineResponse = Extractor.put(LINE_URL + "/" + id, 분당선)
-            .jsonPath().getObject("", LineResponse.class);
+        LineStep.지하철노선을_수정(id, 분당선);
+
         //then
-        assertThat(lineResponse.getName()).isEqualTo(분당선.get("name"));
-        assertThat(lineResponse.getColor()).isEqualTo(분당선.get("color"));
+        LineStep.지하철노선을_확인(id, 분당선);
     }
 
     /**
      * Given 지하철 노선을 생성하고
      * When 생성한 지하철 노선을 삭제하면
      * Then 해당 지하철 노선 정보는 삭제된다
-     * */
+     */
     @DisplayName("지하철노선을 삭제한다.")
     @Test
     void deleteLine() {
 
         //given
-        long id = saveLine(신분당선).jsonPath().getLong("id");
+        Long id = getId(LineStep.지하철노선을_추가(Arrays.asList(신분당선)));
+
+        //when
+        LineStep.지하철노선을_삭제(id);
+
+        //then
+        LineStep.지하철노선_삭제확인(id);
+    }
+
+    /**
+     * Given 지하철역 및 지하철 노선을 생성하고
+     * When 지하철 구간을 추가하면
+     * Then 해당 지하철 노선의 구간의 정보를 확인할 수 있다.
+     */
+    @DisplayName("지하철 구간을 추가한다.")
+    @Test
+    void addSection() {
+
+        //given
+        Map<String, String> params = LineStep.지하철역_및_지하철_노선을_생성(강남역, 판교역, 신분당선);
+
+        //when
+        LineStep.지하철노선의_구간을_추가(params);
+
+        //then
+        LineStep.지하철노선의_구간_확인(params);
+    }
+
+    /**
+     * Given 지하철역 및 지하철 노선을 생성하고 지하철 구간을 추가하고
+     * When 새로운 구간을 연장하면
+     * Then 해당 지하철 노선의 연장된 구간의 정보를 확인할 수 있다.
+     */
+    @DisplayName("지하철 구간을 연장한다.")
+    @Test
+    void addSections() {
+
+        //given
+        Map<String, String> params = LineStep.신분당선_등록();
+
+        //when
+        LineStep.지하철노선의_구간을_연장(params, 신논현역);
+
+        //then
+        LineStep.지하철노선의_구간_확인(params);
+    }
+
+    /**
+     * Given 지하철역 및 지하철 노선을 생성하고 지하철 구간을 추가하고
+     * When 새로운 구간을 등록할 때
+     * Then 새로운 구간의 상행역은 해당 노선에 등록되어있는 하행 종점역이 아닐경우 에러가 발생한다
+     */
```

Comment:
테스트 코드의 when 절과 인수 조건 주석의 when을 일치 시키면 좋을 것 같은데 진우님 생각은 어떠신가요?
```suggestion
    /**
     * Given 지하철역 및 지하철 노선을 생성하고 지하철 구간을 추가하고
     * When 새로운 구간의 상행역은 해당 노선에 등록되어있는 하행 종점역이 아닌 구간을 추가하면
     * Then 구간 등록에 실패한다
     */
```